spark/src/test/java/org/apache/iceberg/spark/data/TestSparkParquetReadMetadataColumns.java

```diff
@@ -77,6 +77,9 @@
   private static final int NUM_ROWS = 1000;
   private static final List<InternalRow> DATA_ROWS;
   private static final List<InternalRow> EXPECTED_ROWS;
+  // We filter for $"data" notStartsWith str1, along the row group splits, to ensure the row group
```

Comment:
This test suite tests the metadata columns, which is unrelated to filtering. I don't think that this should be changed.

It looks like you want a Spark test to validate the end-to-end behavior. I think another suite, or maybe a new one, would be more appropriate.

Reply:
So the reason I originally put this in this file is because there are some tests along the splits.

What I'm looking to test here is that the filter is properly applied to row groups - by checking that the spark parquet reader returns all rows from a given split if any row in that split matches the filter.

I'm going to move this test to its own file.